rules/linux/execution_shell_evasion_linux_binary.toml

```diff
@@ -0,0 +1,107 @@
+[metadata]
+creation_date = "2022/05/06"
+maturity = "production"
+updated_date = "2022/05/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies Linux binary(s) abuse to breakout out of restricted shells or environments by spawning an interactive system
+shell. The linux utility(s) activity of spawning shell is not a standard use of the binary for a user or system
+administrator. It indicates a potentially malicious actor attempting to improve the capabilities or stability of their
+access.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Linux Restricted Shell Breakout via  Linux Binary(s)"
+references = [
+    "https://gtfobins.github.io/gtfobins/apt/",
+    "https://gtfobins.github.io/gtfobins/apt-get/",
+    "https://gtfobins.github.io/gtfobins/nawk/",
+    "https://gtfobins.github.io/gtfobins/mawk/",
+    "https://gtfobins.github.io/gtfobins/awk/",
+    "https://gtfobins.github.io/gtfobins/gawk/",
+    "https://gtfobins.github.io/gtfobins/busybox/",
+    "https://gtfobins.github.io/gtfobins/c89/",
+    "https://gtfobins.github.io/gtfobins/c99/",
+    "https://gtfobins.github.io/gtfobins/cpulimit/",
+    "https://gtfobins.github.io/gtfobins/crash/",
+    "https://gtfobins.github.io/gtfobins/env/",
+    "https://gtfobins.github.io/gtfobins/expect/",
+    "https://gtfobins.github.io/gtfobins/find/",
+    "https://gtfobins.github.io/gtfobins/flock/",
+    "https://gtfobins.github.io/gtfobins/gcc/",
+    "https://gtfobins.github.io/gtfobins/mysql/",
+    "https://gtfobins.github.io/gtfobins/nice/",
+    "https://gtfobins.github.io/gtfobins/ssh/",
+    "https://gtfobins.github.io/gtfobins/vi/",
+    "https://gtfobins.github.io/gtfobins/capsh/",
+    "https://gtfobins.github.io/gtfobins/byebug/",
+]
+risk_score = 47
+rule_id = "52376a86-ee86-4967-97ae-1a05f55816f0"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Execution", "GTFOBins"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
```

Comment:
some improvements we can make:
* more consistency with the shells for each entry - `("bash", "sh", "dash", "zsh")` for all
* same for process.parent.args `"/bin/sh", "/bin/bash", "/bin/dash", "/bin/zsh", "sh", "bash", "dash", "zsh"`
* collapsing logic where shared - many of these look for a shell coming from a parent or parent+arg, we can collapse to make it more maintainable and readable
* use comments to be more descriptive and obvious


Based on all of these changes and suggestions below, the query turns into this

```sql
process where event.type == "start" and
  /* launching shells from unusual parents or parent+arg combos */
  (process.name in ("bash", "sh", "dash","zsh") and
     (process.parent.name == "byebug") or
     (process.parent.name == "crash" and process.parent.args == "-h") or

     /* shells specified in parent args */
     (process.parent.args in ("/bin/sh", "/bin/bash", "/bin/dash", "/bin/zsh", "sh", "bash", "dash", "zsh") and
        (process.parent.name == "nice") or
        (process.parent.name == "cpulimit" and process.parent.args == "-f") or
        (process.parent.name == "find" and process.parent.args == "-exec" and process.parent.args == ";") or
        (process.parent.name == "flock" and process.parent.args == "-u" and process.parent.args == "/")
     ) or

     /* shells specified in args */
     (process.args in ("/bin/sh", "/bin/bash", "/bin/dash", "/bin/zsh", "sh", "bash", "dash", "zsh") and
        (process.name == "capsh" and process.args == "--") or
        (process.name == "sensible-pager" and process.parent.name in ("apt", "apt-get") and process.parent.args == "changelog") or
        (process.name == "busybox" and process.args_count == 2) or
        (process.name == "env" and process.args_count == 2)
     )
  ) or

  (process.parent.name in ("vi", "vim") and process.parent.args == "-c" and process.parent.args in (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")) or
  (process.parent.name in ("c89","c99", "gcc") and process.parent.args in ("sh,-s", "bash,-s", "dash,-s", "zsh,-s", "/bin/sh,-s", "/bin/bash,-s", "/bin/dash,-s", "/bin/zsh,-s") and process.parent.args == "-wrapper") or
  (process.parent.name == "expect" and process.parent.args == "-c" and process.parent.args in ("spawn /bin/sh;interact", "spawn /bin/bash;interact", "spawn /bin/dash;interact", "spawn sh;interact", "spawn bash;interact", "spawn dash;interact")) or
  (process.parent.name == "mysql" and process.parent.args == "-e" and process.parent.args : ("\\!*sh", "\\!*bash", "\\!*dash", "\\!*/bin/sh", "\\!*/bin/bash", "\\!*/bin/dash")) or
  (process.parent.name == "ssh" and process.parent.args == "-o" and process.parent.args in ("ProxyCommand=;sh 0<&2 1>&2", "ProxyCommand=;bash 0<&2 1>&2", "ProxyCommand=;dash 0<&2 1>&2", "ProxyCommand=;/bin/sh 0<&2 1>&2", "ProxyCommand=;/bin/bash 0<&2 1>&2", "ProxyCommand=;/bin/dash 0<&2 1>&2")) or
  (process.parent.name in ("nawk", "mawk", "awk", "gawk") and process.parent.args : "BEGIN {system(*)}")
```

But assessing this approach as a whole reveals that this is just trying to identify suspicious parent-child relationships with a shell coming from an unexpected process (in this case to break out of something). The problem with specifying args and params to this degree is it is a very brittle rule - making it easy to bypass and difficult to maintain.

We could simplify this and reassess the rule based on the relationship by converting the logic to

```sql
process where event.type == "start" and process.name in ("bash", "sh", "dash","zsh") and
  process.parent.name in ("byebug",
                          "crash",
                          "nice",
                          "cpulimit",
                          "find",
                          "flock",
                          "capsh",
                          "sensible-pager",
                          "busybox",
                          "env",
                          "vi", "vim",
                          "c88", "c89"
                          "gcc",
                          "expect",
                          "mysql",
                          "ssh",
                          "nawk", "mawk", "awk", "gawk"
                         )
```

This focuses purely on the relationships. Now, if something seems off here, or too loose or FP, that is probably a good indicator that it may not be a great fit for this rule. 

This also fits well with users adding custom exceptions. I would strongly suggest adding a blurb to `false_positives` as well as a thorough investigation guide to add this context.